app/scripts/store/dimStoreItem.directive.js

```diff
@@ -67,11 +68,26 @@
           ngDialog.closeAll();
 
           if (!dimLoadoutService.dialogOpen) {
-            var bottom = ($(element).offset().top < 300) ? ' move-popup-bottom' : '';
-            var right = ((($('body').width() - $(element).offset().left - 320) < 0) ? ' move-popup-right' : '');
+
+            var settings = dimSettingsService.getSettings().$$state.value,
+              verbosePerks = settings.verbosePerks,
+              minOffset = 300 + (item.perks.length * 44 * verbosePerks),
+              perkStateString = verbosePerks ? 'verbose-perks' :'quiet-perks',
+
+              bottom = ($(element).offset().top < minOffset) ? ' move-popup-bottom' : '',
+              right = ((($('body').width() - $(element).offset().left - 320) < 0) ? ' move-popup-right' : ''),
+              templateString =
+                `
```

Comment:
@lostinplace the main concern is that we have some branches open that touch some of this code directly. Might be a bit of a nightmare to merge them once we are ready to do so. 

I'm all for moving the codebase forward, but this is something that would need to be coordinated across the board for each of the files. This is why we are- at least for now- holding off on some of these refactors and saving them for v4.